apps/api/src/app/widgets/dtos/feeds-response.dto.ts

```diff
@@ -0,0 +1,131 @@
+import { ApiExtraModels, ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { ActorTypeEnum, ChannelTypeEnum, IActor, INotificationDto } from '@novu/shared';
+
+import { SubscriberResponseDto } from '../../subscribers/dtos';
+import { EmailBlock, MessageCTA } from './message-response.dto';
+
+class Actor implements IActor {
+  @ApiProperty()
+  data: string | null;
+
+  @ApiProperty({ enum: ActorTypeEnum })
+  type: ActorTypeEnum;
+}
+
+@ApiExtraModels(EmailBlock, MessageCTA)
+export class NotificationDto implements INotificationDto {
+  @ApiPropertyOptional()
+  _id: string;
+
+  @ApiProperty()
+  _templateId: string;
+
+  @ApiProperty()
+  _environmentId: string;
+
+  @ApiProperty()
+  _messageTemplateId: string;
+
+  @ApiProperty()
+  _organizationId: string;
+
+  @ApiProperty()
+  _notificationId: string;
+
+  @ApiProperty()
+  _subscriberId: string;
+
+  @ApiProperty()
+  _feedId: string;
+
+  @ApiProperty()
+  _jobId: string;
+
+  @ApiPropertyOptional()
+  createdAt: string;
+
+  @ApiPropertyOptional()
+  updatedAt: string;
+
+  @ApiPropertyOptional()
+  expireAt: string;
+
+  @ApiPropertyOptional({
+    type: Actor,
+  })
+  actor?: Actor;
+
+  @ApiPropertyOptional({
+    type: SubscriberResponseDto,
+  })
+  subscriber?: SubscriberResponseDto;
+
+  @ApiProperty()
+  transactionId: string;
+
+  @ApiPropertyOptional()
+  templateIdentifier: string;
+
+  @ApiPropertyOptional()
+  providerId: string;
+
+  @ApiProperty()
+  content: string;
+
+  @ApiProperty()
+  subject?: string;
+
+  @ApiProperty({
+    enum: ChannelTypeEnum,
+  })
+  channel: ChannelTypeEnum;
+
+  @ApiProperty()
+  read: boolean;
+
+  @ApiProperty()
+  seen: boolean;
+
+  @ApiProperty()
+  deleted: boolean;
+
+  @ApiPropertyOptional()
+  deviceTokens?: string[];
+
+  @ApiProperty({
+    type: MessageCTA,
+  })
+  cta: MessageCTA;
+
+  @ApiProperty({
+    enum: ['sent', 'error', 'warning'],
+  })
+  status: 'sent' | 'error' | 'warning';
```

Comment:
Same question for these statuses. They feel a bit generic in terms of DX raising a question of how a developer can use them.

Reply:
Some of the fields are not defined in the Notification class, this DTO is just the current state.